### PR TITLE
feat: fusd

### PR DESCRIPTION
**Pull request by mul53**
Changes
- [x] Users can withdraw tokens in proportionate amounts when system in depegged

### CODE REVIEW

File: contracts/masset/Masset.sol [hunk 1]
```diff
@@ -697,7 +684,7 @@ contract Masset is
     function collectInterest()
         external
         override
-        onlySavingsManager
+        onlyGovernor
```
Comment (leonprou) [inline contracts/masset/Masset.sol]: let's see if we can keep the savingManager for future. 
For now we can put the governor and multisig as a same account

File: contracts/masset/Masset.sol [hunk 2]
```diff
@@ -431,7 +418,7 @@ contract Masset is
         uint256 _mAssetQuantity,
         uint256[] calldata _minOutputQuantities,
         address _recipient
-    ) external override nonReentrant whenNoRecol returns (uint256[] memory outputQuantities) {
+    ) external override nonReentrant returns (uint256[] memory outputQuantities) {
```
Comment (leonprou) [inline contracts/masset/Masset.sol:434]: Let's think better on this scenario.
if a token is depegging a user can withdraw the token, and then when the token is back the user  just gets more tokens, and it's less tokens in the pool.

Comment (leonprou) [reply contracts/masset/Masset.sol:434]: Let's add more tests to this scenario, and let's write a test of manipulation of this flow: 
1. token price drops, 2. deposit -> withdraw loop untill the depeg is realized

I suggest to start with a close range of weights to we can minimize this affect.

**Review by leonprou — COMMENTED** [review]
Let's also take into account the ERC677 tokens that gonna be used

File: contracts/masset/Masset.sol (hunk 2 above)
Comment (mul53) [reply contracts/masset/Masset.sol:434]: @leonprou when we pause the contracts when the token price drops, users won't be allowed to deposit and swap but only withdraw proportionate amounts, so they can't arbitrage the pool. We then unpause the contract when the contract when the token gets back to peg.

**Pull request merged**